File: Exercise_1/Main.py
class Vehicle:
    def __init__(self, brand = "Unknown", year = 0):
        self.__brand = brand
        self.__year = year

    def getBrand(self): return self.__brand
    def getYear(self): return self.__year

    def setBrand(self, b): self.__brand = b
    def setYear(self, y): self.__year = y

    def displayInfo(self):
        print(f"Brand: {self.__brand}, Year: {self.__year}")

class Car(Vehicle):
    def __init__(self, brand="Unknown", year=0, model ="Unknown"):
        super().__init__(brand, year)
        self.__model = model

    def displayInfo(self):
        print(f"Brand: {self.getBrand()}, Model: {self.__model}, Year: {self.getYear()}")

class Bike(Vehicle):
    def __init__(self, brand="Unknown", year=0, engineCapacity=0):
        super().__init__(brand, year)
        self.__engineCapacity = engineCapacity

    def displayInfo(self):
        print(f"Brand: {self.getBrand()}, Engine: {self.__engineCapacity}cc, Year: {self.getYear()}")

# 4. Main Execution
if __name__ == "__main__":
    print("--- Creating car1 (Default) ---")
    car1 = Car() 
    car1.displayInfo()

    print("\n--- Creating car2 (Parameterized) ---")
    car2 = Car("Toyota", 2023, "Camry")
    car2.displayInfo()

    print("\n--- Creating bike1 (Default) ---")
    bike1 = Bike()
    bike1.displayInfo()

    print("\n--- Creating bike2 (Parameterized) ---")
    bike2 = Bike("Yamaha", 2020, 150)
    bike2.displayInfo()

File: Exercise_1/Main.java
package Exercise_1;

class Vehicle {
    private String brand;
    private int year;

    public Vehicle() {
        this.brand = "Unknown";
        this.year = 0;
    }

    public Vehicle(String b, int y) {
        this.brand = b;
        this.year = y;
    }

    public String getBrand() { return brand; }
    public int getYear() {return year;}

    public void setBrand(String b) {brand = b;}
    public void setYear(int y) {year = y;}

    public void displayInfo() {
        System.out.println("Brand: " + brand + ", Year: " + year);
    }
}

class Car extends Vehicle{
    private String model;

    public Car() {
        super();
        this.model = "Unknown";
    }

    public Car(String brand, int year, String model) {
        super(brand, year);
        this.model = model;
    }

    @Override
    public void displayInfo() {
        System.out.println("Brand: " + getBrand() + ", Model: " + model + ", Year: " + getYear());
    }
}

class Bike extends Vehicle {
    private int engineCapacity;

    public Bike() {
        super();
        this.engineCapacity = 0;
    }

    public Bike(String brand, int year, int engineCapacity) {
        super(brand, year);
        this.engineCapacity = engineCapacity;
    }

    @Override
    public void displayInfo() {
        System.out.println("Brand: " + getBrand() + ", Engine: " + engineCapacity + "cc, Year: " + getYear());
    }
}

public class Main {
    public static void main(String[] args) {
        System.out.println("--- Creating car1 (Default) ---");
        Car car1 = new Car();
        car1.displayInfo();

        System.out.println("\n--- Creating car2 (Parameterized) ---");
        Car car2 = new Car("Toyota", 2023, "Camry");
        car2.displayInfo();

        System.out.println("\n--- Creating bike1 (Default) ---");
        Bike bike1 = new Bike();
        bike1.displayInfo();

        System.out.println("\n--- Creating bike2 (Parameterized) ---");
        Bike bike2 = new Bike("Yamaha", 2020, 150);
        bike2.displayInfo();
    }
}